### PR TITLE
Added steamcmd build container for first build step

### PR DESCRIPTION
**Pull request by mikecarey134**

### CODE REVIEW

File: docker/valheim/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,17 @@
-FROM steamcmd/steamcmd:debian-12
-WORKDIR /Valheim
+FROM ubuntu:23.10
```
Comment (lamartina) [inline docker/valheim/Dockerfile:1]: Why, the switch to Ubuntu from debian?

File: docker/valheim/run.sh [hunk 2]
```diff
@@ -1 +1 @@
-docker run -d -p 2456:2456 -v valheim:/Valheim --name valheim valheim
+docker run -d -p 2456:2458 -v valheim:/server --name valheim valheim
```
Comment (lamartina) [inline docker/valheim/run.sh]: Different port?

**Review by lamartina — CHANGES_REQUESTED** [review]
Added comments.

File: docker/valheim/Dockerfile (hunk 1 above)
Comment (mikecarey134) [reply docker/valheim/Dockerfile:1]: was testing functionality of steamcmd so far I can get the server running but cant connect to it

File: docker/valheim/run.sh (hunk 2 above)
Comment (mikecarey134) [reply docker/valheim/run.sh]: range 2456-2458

Comment (mikecarey134) [reply docker/valheim/run.sh]: nvm I fucked this up lol

Comment (mikecarey134) [reply docker/valheim/run.sh]: fixed this in latest push

Comment (lamartina) [reply docker/valheim/run.sh]: Looks better now lol

**Review by mikecarey134 — COMMENTED** [review]
Updated for your review

**Review by lamartina — APPROVED** [review]
Loooooks guuda like cheese

**Review by lamartina — COMMENTED** [review]
Looks good

**Pull request merged**